sys/freebsd/dev_crypto.txt

```diff
@@ -0,0 +1,68 @@
+# Copyright 2020 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+include <sys/time.h>
+include <opencrypto/cryptodev.h>
+
+resource fd_crypto[fd]
+
+openat$crypto(fd const[AT_FDCWD], file ptr[in, string["/dev/crypto"]], flags flags[open_flags], mode const[0]) fd_crypto
+
+ioctl$CIOCFINDDEV(fd fd_crypto, cmd const[CIOCFINDDEV], arg ptr[in, crypt_find_op])
+ioctl$CIOCGSESSION(fd fd_crypto, cmd const[CIOCGSESSION], arg ptr[in, session_op])
+ioctl$CIOCGSESSION2(fd fd_crypto, cmd const[CIOCGSESSION2], arg ptr[in, session2_op])
+ioctl$CIOCCRYPT(fd fd_crypto, cmd const[CIOCCRYPT], arg ptr[in, crypt_op])
+ioctl$CIOCCRYPTAEAD(fd fd_crypto, cmd const[CIOCCRYPTAEAD], arg ptr[in, crypt_aead])
+ioctl$CIOCFSESSION(fd fd_crypto, cmd const[CIOCFSESSION], arg ptr[in, array[int32]])
```

Comment:
There is another compat ioctl that would be worth covering. It is CRIOGET. See sys/opencrypto/cryptodev.c. It's kind of weird in that it returns a fd_cryptofd using the pointer argument.

Since the value of CRIOGET is not defined in a header, you'll have to hard-code it here. syzlang has a "define" statement for that purpose.

Reply:
I have added it. I have taken CRIOGET value to be 100 as passed in cryptodev.c

Reply:
100 is not quite right. Look at the values for other ioctls in dev_crypto.txt.const and look at the definition of `_IOWR` in sys/ioccom.h. I believe the value will be `0xc0000000 | ((sizeof(uint32) << 16) | ('c' << 8) | 100`.

Reply:
Okay. I understand this now `(inout) | (((len) & IOCPARM_MASK) << 16) | ((group) << 8) | (num)`